src/DynamoCoreWpf/Extensions/IViewExtension.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo.Extensions;
+
+namespace Dynamo.Wpf.Extensions
+{
+    /// <summary>
+    /// An extension to the UI layer of Dynamo.
+    /// </summary>
+    public interface IViewExtension : IDisposable
```

Comment:
All for readability (have had enough fun with `class SearchLibrary<TEntry, TItem> : SearchDictionary<TEntry>, ISource<TItem>  where TEntry : ISearchEntry, ISource<TItem>`, which I thought is hard to reason about).